src/dist/replication/lib/replica_stub.cpp

```diff
@@ -1216,14 +1221,18 @@ void replica_stub::close()
         while (_closing_replicas.empty() == false)
         {
             task_ptr task = _closing_replicas.begin()->second.first;
+            global_partition_id tmp_gpid = _closing_replicas.begin()->first;
             _replicas_lock.unlock();
 
             task->wait();
 
-
             _counter_replicas_closing_count.decrement();
             _replicas_lock.lock();
-            _closing_replicas.erase(_closing_replicas.begin());
+            // task will automatically remove this replica from _closing_replicas
+            if(false == _closing_replicas.empty())
```

Comment:
the above "_counter_replicas_closing_count.decrement();" should also be removed to avoid double decrement.